src/commands/ddev/init.tsx

```diff
@@ -0,0 +1,150 @@
+    "This command initializes a new ddev configuration for an existing app installation in the current directory.\n" +
+    "\n" +
+    "More precisely, this command will do the following:\n\n" +
+    "  1. Create a new ddev configuration file in the .ddev directory, appropriate for the reference app installation\n" +
+    "  2. Initialize a new ddev project with the given configuration\n" +
+    "  3. Install the official mittwald DDEV addon\n" +
+    "  4. Add SSH credentials to the DDEV project\n" +
+    "\n" +
+    "This command can be run repeatedly to update the DDEV configuration of the project.\n" +
+    "\n" +
+    "Please note that this command requires DDEV to be installed on your system.";
+
+  static flags = {
+    ...processFlags,
+    "project-name": Flags.string({
+      summary: "DDEV project name",
+      description: "The name of the DDEV project",
+      required: false,
+      default: undefined,
+    }),
+    "override-mittwald-plugin": Flags.string({
+      summary: "override the mittwald plugin",
+      helpGroup: "Development",
+      description:
+        "This flag allows you to override the mittwald plugin that should be installed by default; this is useful for testing purposes",
+      default: "mittwald/ddev",
+    }),
+  };
+  static args = {
+    ...appInstallationArgs,
+  };
+
+  protected async exec(): Promise<void> {
+    const appInstallationId = await this.withAppInstallationId(Init);
+    const r = makeProcessRenderer(this.flags, "Initializing DDEV project");
+
+    await assertDDEVIsInstalled(r);
+
+    const config = await this.writeMittwaldConfiguration(r, appInstallationId);
+    const projectName = await this.determineProjectName(r);
+
+    await this.initializeDDEVProject(r, config, projectName);
+    await this.installMittwaldPlugin(r);
+    await this.addSSHCredentials(r);
+
+    await r.complete(<DDEVInitSuccess />);
+  }
+
+  protected render(): React.ReactNode {
+    return undefined;
+  }
+
+  private async addSSHCredentials(r: ProcessRenderer) {
+    await spawnInProcess(r, "adding SSH credentials to DDEV", "ddev", [
+      "auth",
+      "ssh",
+    ]);
+  }
+
+  private async installMittwaldPlugin(r: ProcessRenderer) {
+    const { "override-mittwald-plugin": mittwaldPlugin } = this.flags;
+    await spawnInProcess(r, "installing mittwald plugin", "ddev", [
+      "get",
+      mittwaldPlugin,
+    ]);
+  }
+
+  private async initializeDDEVProject(
+    r: ProcessRenderer,
+    config: Partial<DDEVConfig>,
+    projectName: string,
+  ): Promise<void> {
+    await spawnInProcess(r, "initializing DDEV project", "ddev", [
+      "config",
+      "--project-name",
+      projectName,
+      ...ddevConfigToFlags(config),
+    ]);
+  }
+
+  private async determineProjectName(r: ProcessRenderer): Promise<string> {
+    const { "project-name": projectName } = this.flags;
+    if (projectName) {
+      return projectName;
+    }
+
+    return await r.addInput("Enter the project name", false);
```

Comment:
Improvement idea (maybe another PR): Use Dirname as default value. So if an empty name is given, the directory name is used.

Reply:
Partially unrelated, but that reminds me: I actually wanted to add a lookup to re-use the _existing_ project name, when the command is run in a directory where a DDEV project is already initialized...

Reply:
Regarding the default value: I wanted to mirror DDEV's behaviour here (which also prompts the user if no name is specified), as to not break anyone's expectations.

Reply:
yes, I do agree with the behaviour to ask for a name if none is given. But what happens if I just hit enter when the input query is shown? Does it set an empty name? In this case the dirname could be a sensible fallback, if also showed as default.

Reply:
Created #230 as follow-up